src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -130,12 +130,102 @@ export const currentChannelDisplayableMessages = createSelector(
         id: message.id,
         type: message.type,
         message: message.message,
-        createdAt: formatMessageDisplayDate(message.createdAt),
+        createdAt: message.createdAt,
         nickname: user.username,
       };
     })
 );
 
+// returns array of messages SORTED by createdAt
+export const currentChannelMessagesSortedByTime = createSelector(
+  currentChannelDisplayableMessages,
+  (messages) =>
+    messages.sort((a, b) => b.createdAt - a.createdAt)
+);
+
+// returns array of messages SORTED and GROUPED by createdAt and user
+export const currentChannelMessagesGroupedByUserTime = createSelector(
+  currentChannelMessagesSortedByTime,
+  (messages) => {
+    const timeOfGroupingMessages = 300
+    let newMessages = []
+
+    for (let indexOfMessages = 0; indexOfMessages < messages.length; indexOfMessages++) {
+      let currentMessage = messages[indexOfMessages]
+
+      if (messages[indexOfMessages + 1]) {
+        while ((currentMessage.createdAt - messages[indexOfMessages + 1].createdAt) < timeOfGroupingMessages
+          && currentMessage.nickname === messages[indexOfMessages + 1].nickname) {
+          currentMessage = {
+            ...currentMessage,
+            message: currentMessage.message + "\n" + messages[indexOfMessages + 1].message
+          }
+          indexOfMessages++
+        }
+      }
+
+      newMessages.push(currentMessage)
+    }
+    return newMessages
+  }
+);
+
+// returns array of 'day' object with day and grouped messages
+export const currentChannelMessagesGroupedByDay = createSelector(
+  currentChannelMessagesGroupedByUserTime,
+  (messages) => {
+    let messagesByDay: Array<{
+      day: string,
+      messages: any[]
+    }> = []
+
+    for (const message of messages) {
```

Comment:
Similar case as above. I'm afraid of the for loop in this place, especially there is a fine designed `reduce` grouping in ZbayLite's ChannelMessages component

Reply:
as above